File: 백준/Silver/10866. 덱/덱.py
import sys
input = sys.stdin.readline().rstrip()

N = int(input)

deque = []

def push_front(X) :
    global deque
    deque = deque[::-1]
    deque.append (X)
    deque = deque[::-1]
    
def push_back(X) : 
    deque.append (X)
    
def pop_front() : 
    if deque == [] :
        print (-1)
    
    else :    
        print (deque[0])
        deque.pop(0)
        
def pop_back() :
    if deque == [] :
        print (-1)
    
    else :    
        print (deque[-1])
        deque.pop(-1)
        
def size() :
    print (len(deque))
    
def empty() :
    if deque == [] :
        print (1)
    
    else :    
        print (0)
        
def front() :
    if deque == [] :
        print (-1)
    
    else :    
        print (deque[0])
        
def back() :
    if deque == [] :
        print (-1)
    
    else :    
        print (deque[-1])
        
for i in range (N) :
    input = sys.stdin.readline().rstrip()
    operation = input
    
    if 'push_front' in operation : 
        push_front(int(operation.split(' ')[1]))
        
    elif 'push_back' in operation : 
        push_back(int(operation.split(' ')[1]))
        
    elif operation == 'pop_front' : 
        pop_front()
        
    elif operation == 'pop_back' : 
        pop_back()
        
    elif operation == 'size' : 
        size()
        
    elif operation == 'empty' : 
        empty()
        
    elif operation == 'front' : 
        front()
        
    elif operation == 'back' : 
        back()
        

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <cstdio>
#include <deque>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);
    
    std::deque<int> deque;
    long long int N;
    long long int counting = 0;
    
    std::cin >> N;
    
    for (int i = 0; i < N; i++) {
        std::string command;
        std::cin >> command;

        if (command.compare("push_front") == 0) {
            long long int X;
            std::cin >> X;
            
            deque.push_front(X);
            counting += 1;
        }
        
        if (command.compare("push_back") == 0) {
            long long int X;
            std::cin >> X;
            
            deque.push_back(X);
            counting += 1;
        }
        
        if (command.compare("pop_front") == 0) {
            if (counting == 0) {
                std::cout << -1 << "\n";
            }
            
            else {
                std::cout << deque.front() << "\n";
                deque.pop_front();
                counting -= 1;
            }
        }
        
        if (command.compare("pop_back") == 0) {
            if (counting == 0) {
                std::cout << -1 << "\n";
            }
            
            else {
                std::cout << deque.back() << "\n";
                deque.pop_back();
                counting -= 1;
            }
        }
        
        if (command.compare("size") == 0) {
            std::cout << counting << "\n";
        }
        
        if (command.compare("empty") == 0) {
            if (counting == 0) {
                std::cout << 1 << "\n";
            }
            
            else {
                std::cout << 0 << "\n";
            }
        }
        
        if (command.compare("front") == 0) {
            if (counting == 0) {
                std::cout << -1 << "\n";
            }
            
            else {
                std::cout << deque.front() << "\n";
            }
        }
        
        if (command.compare("back") == 0) {
            if (counting == 0) {
                std::cout << -1 << "\n";
            }
            
            else {
                std::cout << deque.back() << "\n";
            }
        }
    }
    
    return 0;
}